services/QuillLMS/app/controllers/cms/unit_templates_controller.rb

```diff
@@ -28,23 +28,26 @@ def new
   end
 
   def create
-    @unit_template = UnitTemplate.new(unit_template_params)
+    @unit_template = UnitTemplate.new(unit_template_params.except(:activity_ids))
     if @unit_template.save!
+      unit_template_params[:activity_ids].each_with_index do |activity_id, order_number|
+        ActivitiesUnitTemplate.create(activity_id:, order_number:, unit_template_id: @unit_template.id)
+      end
+
       render json: @unit_template
     else
       render json: {errors: @unit_template.errors}, status: 422
     end
   end
 
   def update
-    params = unit_template_params
-    @unit_template.activities = []
-    @unit_template.save
-    if @unit_template.update(params)
-      @unit_template.activities_unit_templates.each do |aut|
-        order_number = params['activity_ids'].index(aut.activity_id)
-        aut.update(order_number: order_number)
+    if @unit_template.update(unit_template_params.except(:activity_ids))
+      @unit_template.activities_unit_templates.destroy_all
```

Comment:
Destroying and recreating records like this feels a little awkward, but given how rarely staff touch this data, I can't imagine that this is worth being particularly worried about.